app/assets/javascripts/infinity_scroll.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+function init_infinity_scroll(initial_page_number, request_url, load_first_page) {
+    var page_number;
+    if (initial_page_number === undefined) {
+        page_number = 1;
+    }
+    else {
+        page_number = initial_page_number
+    }
+    if (load_first_page === true) load_data();
+    start_infinity_scroll();
+
+    function start_infinity_scroll() {
+        $(window).on('scroll', function () {
+            if (end_of_page()) {
+                load_data();
+                $(window).off('scroll');
+            }
+        });
+    }
+
+    function end_of_page() {
+        return $(window).scrollTop() + $(window).height() > $(document).height() - 100
+    }
+
+    function load_data() {
```

Comment:
ALL this js could be removed if you use this gem
https://github.com/kaminari/kaminari/wiki/How-To:-Create-Infinite-Scrolling-with-jQuery
OR others like https://github.com/magoosh/jquery-infinite-pages
...
it's always better to check if there are any gem exists for your task